internal/pkg/keeper/client.go

```diff
@@ -0,0 +1,333 @@
+	if err != nil {
+		return fmt.Errorf("failed to check the %s service registry status: %s", k.serviceKey, err.Error())
+	}
+
+	// call the PUT registry API to update the registry if the service already exists
+	// otherwise, call the POST API to create the registry
+	httpMethod := http.MethodPost
+	if resp.StatusCode == http.StatusOK {
+		httpMethod = http.MethodPut
+	}
+	req, err := http.NewRequest(httpMethod, k.config.GetRegistryUrl()+ApiRegisterRoute, bytes.NewReader(jsonEncodedData))
+	if err != nil {
+		return fmt.Errorf("failed to create register request: %s", err.Error())
+	}
+	req.Header.Set(ContentType, ContentTypeJSON)
+
+	client := http.Client{Timeout: defaultTimeout}
+	resp, err = client.Do(req)
+	if err != nil {
+		return fmt.Errorf("http error: %s", err.Error())
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusNoContent {
+		var response BaseResponse
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response body: %s", err.Error())
+		}
+		err = json.Unmarshal(bodyBytes, &response)
+		if err != nil {
+			return fmt.Errorf("failed to decode response body: %s", err.Error())
+		}
+		return fmt.Errorf("failed to register %s: %s", k.serviceKey, response.Message)
+	}
+
+	return nil
+}
+
+func (k *keeperClient) Unregister() error {
+	registrationReq := AddRegistrationRequest{
+		BaseRequest: BaseRequest{
+			Versionable: Versionable{ApiVersion: ApiVersion},
+		},
+		Registration: RegistrationDTO{
+			ServiceId: k.serviceKey,
+			Host:      k.serviceHost,
+			Port:      k.servicePort,
+			HealthCheck: HealthCheck{
+				Interval: k.healthCheckInterval,
+				Path:     k.healthCheckRoute,
+				Type:     "http",
+			},
+			Status: haltStatus,
+		},
+	}
+
+	jsonEncodedData, err := json.Marshal(registrationReq)
+	if err != nil {
+		return fmt.Errorf("failed to encode unregister request: %w", err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, k.config.GetRegistryUrl()+ApiRegisterRoute, bytes.NewReader(jsonEncodedData))
+	if err != nil {
+		return fmt.Errorf("failed to create unregister request: %w", err)
+	}
+
+	client := http.Client{Timeout: defaultTimeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusNoContent {
+		var response BaseResponse
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response body: %w", err)
+		}
+		err = json.Unmarshal(bodyBytes, &response)
+		if err != nil {
+			return fmt.Errorf("failed to decode response body: %w", err)
+		}
+		return fmt.Errorf("failed to unregister %s: %s", k.serviceKey, response.Message)
+	}
+
+	return nil
+}
+
+func (k *keeperClient) RegisterCheck(id string, name string, notes string, url string, interval string) error {
+	// keeper combines service discovery and health check into one single register request
```

Comment:
This needs to be implemented. The are users that have added additional RegisterChecks via this API

Reply:
Keeper does not support this for now, may be an enhancement in the future.

Reply:
That is a breaking change then as this is current functionality when using Consul. @cloudxxx8 FYI.

Reply:
@lenny-intel it's not really breaking anything
core-keeper requires the healthcheck setting when the registration creation
https://github.com/edgexfoundry/go-mod-core-contracts/blob/7c7fc6d4916a7b9a2c91d36c929726a20a3654c9/dtos/registration.go#L20

Consul requires a separated call to this RegisterCheck, but there is no impact if core-keeper does nothing in RegisterCheck func